daemon/src/model/cfd.rs

```diff
@@ -2371,6 +2390,212 @@ mod tests {
         assert_eq!(funding_fee.rate, funding_rate);
     }
 
+    #[test]
+    fn given_collab_settlement_then_cannot_start_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok());
```

Comment:
The test name doesn't mention that we are testing the scenario of "rollover started - collab settlement started - rollover started".

Remove this?

Reply:
it's more a sanity check that we are able to start rollover *before* we do whatever we do. But I can remove them. This goes towards "a test should only test one thing". Sometimes I find it useful to show to the reader that "at this point we can, then we do this, then we can't..." 

Don't mind much though